### ISSUE TITLE
Bug in Tensorflow implementations

### ISSUE THREAD

**Issue by rlronan** [suragnair/alpha-zero-general#159]
I think there may be a bug in the Tensorflow implementations of OthelloNNet, and its derivatives. 
In the last layer of policy network, the output is evaluated without being put through a softmax layer. In [line 36:](https://github.com/suragnair/alpha-zero-general/blob/master/othello/tensorflow/OthelloNNet.py#L36) we take the net output as self.pi. In line 37 we take the softmax of that as self.probs: 

```
            self.pi = Dense(s_fc2, self.action_size)                                                        # batch_size x self.action_size
            self.prob = tf.nn.softmax(self.pi)
            self.v = Tanh(Dense(s_fc2, 1))                                                               # batch_size x 1

            self.calculate_loss()
```
But in calculate_loss we compare self.target_pis and self.pi, rather than self.target_pis and **self.probs**:
```
    def calculate_loss(self):
        self.target_pis = tf.placeholder(tf.float32, shape=[None, self.action_size])
        self.target_vs = tf.placeholder(tf.float32, shape=[None])
        self.loss_pi =  tf.losses.softmax_cross_entropy(self.target_pis, self.pi)
        self.loss_v = tf.losses.mean_squared_error(self.target_vs, tf.reshape(self.v, shape=[-1,]))
        self.total_loss = self.loss_pi + self.loss_v
        update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
        with tf.control_dependencies(update_ops):
            self.train_step = tf.train.AdamOptimizer(self.args.lr).minimize(self.total_loss)
```

I think we should be comparing target pis and self.probs.

**Comment by rlronan** [comment]
Or is prob only supposed to be used in Predict? Because then this is not a bug, and I will delete it.

**Comment by rlronan** [comment]
Oh right duh, this is handled by the softmax_cross_entropy loss. My apologies!
